### DISCUSSION TITLE
repeat_every use

### DISCUSSION THREAD

**Discussion by dor1202** [fastapi/fastapi#6000] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@repeat_every(seconds=5)
def remove_expired_tokens_task() -> None:
    print("test")
```

### Description

How do I bind the Repeated Tasks to the server?
I know its a simple thing but I couldn't find doc for binding this function to the server.

### Operating System

macOS

### FastAPI Version

0.88.0

### Python Version

3.10.9

**Comment by dor1202** [comment]
Am I supposed to bind the function to a `@app.on_event("startup")`?

**Comment by clemens-tolboom** [comment]
I have
```python
    app = FastAPI()

    @app.on_event("startup")
    @repeat_every(seconds=0.01, wait_first=True)
    async def poll_from_gui():
```

order matters ... see fastapi_utils issue queue.

**Comment by williamjamir** [comment, marked as answer]
It looks like `@repeat_every` is from the `fastapi_utils` package.
Perhaps raising this question on the repository will bring different answers.

That being said, according to the documentation for the repeat_every decorator (https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/#the-repeat_every-decorator), the loop will only start when the decorated function is called. 

In other words, if you want to bind the repeated task to the server, you need to call the function at startup.

To do this, you can decorate the function with the `@app.event("startup")` decorator.
This will ensure that the function is called during server startup and will then be called repeatedly while the server is still running.

Here's an example:

```python
from fastapi import FastAPI
from fastapi_utils.tasks import repeat_every

app = FastAPI()

@repeat_every(seconds=5)
async def remove_expired_tokens_task() -> None:
    print("test")

@app.on_event("startup")
async def startup_event():
    await remove_expired_tokens_task()
```

In this example, the `remove_expired_tokens_task` function will be called every 5 seconds starting from server startup. 


I hope this helps!

**Reply by dor1202** [reply]
@williamjamir Thank you for the response,
That was exactly what I was looking for! dividing the bind and the repeat.
